packages/rest/src/rest-http-error.ts

```diff
@@ -12,7 +13,33 @@ export namespace RestHttpErrors {
     const msg = `Parameters with "in: ${location}" are not supported yet.`;
     return new HttpErrors.NotImplemented(msg);
   }
-  export function invalidRequestBody(msg: string): HttpErrors.HttpError {
-    return new HttpErrors.UnprocessableEntity(msg);
+  export const INVALID_REQUEST_BODY_MESSAGE =
+    'The request body is invalid. See error object `details` property for more info.';
+  export function invalidRequestBody(): HttpErrors.HttpError {
+    return new HttpErrors.UnprocessableEntity(INVALID_REQUEST_BODY_MESSAGE);
+  }
+  /**
+   * An invalid request body error contains a `details` property as the machine-readable error.
+   * Each entry in `error.details` contains 4 attributes: `path`, `code`, `info` and `message`.
+   * `ErrorDetails` defines the type of each entry, which is an object.
+   * The type of `error.details` is `ErrorDetails[]`.
+   */
+  export interface ErrorDetails {
+    /**
+     * A path to the invalid field.
+     */
+    path: string;
+    /**
+     * A single word code represents the error's type.
+     */
+    code: string;
+    /**
+     * Some additional details that the 3 attributes above don't cover.
+     */
+    info: object;
+    /**
+     * A humnan readable description of the error.
+     */
+    message?: string;
```

Comment:
>   // Excluded if messages set to false.
>    message?: string;

IIUC, we don't set `messages: false` in AJV options, therefore AJV will always fill the message for us and this property is always set.

I am proposing to keep `ErrorDetails#message` as always provided to make it easier for our clients to handle validation errors. Thoughts?

Reply:
sounds good reverted it to a required field.